### PR TITLE
Add pebble metrics

### PR DESCRIPTION
**Pull request by ischasny**

### CODE REVIEW

File: metrics/metrics.go [hunk 1]
```diff
@@ -63,6 +72,10 @@ func (m *Metrics) Start(_ context.Context) error {
 
 	go func() { _ = m.s.Serve(mln) }()
 
+	if m.pebbleMetrics != nil {
+		m.pebbleMetrics.start()
```
Comment (masih) [inline metrics/metrics.go:76]: Handle error?

File: metrics/pebble_metrics.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	cacheHits asyncint64.Gauge
+	// cacheMisses reports number of cache misses.
+	cacheMisses asyncint64.Gauge
+}
+
+func (pm *pebbleMetrics) start() error {
+	var err error
+
+	if pm.fsyncFlushCount, err = pm.meter.AsyncInt64().Gauge(
+		"ipni/dhstore/pebble/fsync_flush_count",
```
Comment (masih) [inline metrics/pebble_metrics.go:36]: Consistent terminology with pebble?
```suggestion
		"ipni/dhstore/pebble/flush_count",
```

File: metrics/pebble_metrics.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	"go.opentelemetry.io/otel/metric/unit"
+)
+
+// pebbleMetrics asynchronously reports metrics of pebble DB
+type pebbleMetrics struct {
+	metricsProvider func() *pebble.Metrics
+	meter           cmetric.Meter
+
+	// fsyncFlushCount reports the total number of flushes
+	fsyncFlushCount asyncint64.Gauge
```
Comment (masih) [inline metrics/pebble_metrics.go:20]: ditto re consistent terminology

File: metrics/pebble_metrics.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+			pm.cacheMisses,
+		},
+		pm.reportAsyncMetrics,
+	)
+}
+
+func (pm *pebbleMetrics) reportAsyncMetrics(ctx context.Context) {
+	m := pm.metricsProvider()
+
+	pm.fsyncFlushCount.Observe(ctx, m.Flush.Count)
```
Comment (masih) [inline metrics/pebble_metrics.go:90]: Let's see what metrics we get from the existing ones.

Eventually we also might want:
* `ReadAmp()`
* `Total().Bytes*` and ``Total().Tables*` numbers

**Review by masih — APPROVED** [review]

File: metrics/metrics.go (hunk 1 above)
Comment (ischasny) [reply metrics/metrics.go:76]: good point,fixed

File: metrics/pebble_metrics.go (hunk 4 above)
Comment (ischasny) [reply metrics/pebble_metrics.go:90]: Ok, let me add them now

**Pull request merged**